### PR TITLE
fix docker entrypoint

### PR DESCRIPTION
**Pull request by kiorky**

### CODE REVIEW

**Comment by maxidor** [comment]
Please explain the purpose of this, I don't see a functionnal change here, just a cosmetic one.

**Comment by kiorky** [comment]
the -z is bash, and doesnt work with sh ... the entry point then overwrites any custom configuration and prevents the daemon from starting.

**Comment by kiorky** [comment]
And the "starting mxisd" is also in the if block, then if you specified a specific custom conf, you wont see the start statement in logs.

**Comment by maxidor** [comment]
I will have to decline this PR I'm afraid:
- I would like to keep bash compatibility. the shell info should be simply adapted.
- The config file env variable is work in progress and is only used as an internal gimmick currently. It was left undocumented on purpose.
- The `Starting mxisd` is at the right place, as I wanted to distinguish between statements related to config file and actual mxisd logs. If mxisd starts directly, there is no need for the log statement.

Thank you for the effort tho!

**Comment by kiorky** [comment]
Then use an image which uses bash ...........

**Comment by kiorky** [comment]
@maxidor the baseimage you are using does not have bash ;)

**Comment by kiorky** [comment]
Uhm, wait, it have it, it's only that you do not used the correct bash syntax "[" vs "[[" i ll reopen another PR.

**Comment by maxidor** [comment]
All I can tell you is that it works in its current form, I've doubled checked before replying.

**Comment by kiorky** [comment]
and definitly no, the docker image does not have it ..

**Comment by maxidor** [comment]
Indeed, I was sure I was using the stretch variant, and not alpine. Sorry about that, your PR is totally sensible.
So, I would like to keep `Starting mxisd` in the first `if`, but happy to consider to remove all the bash stuff then.

**Comment by kiorky** [comment]
@maxidor arf, didnt saw your message, the pr has now bash(dockerfile change), and the bashisms are fixed.

**Comment by maxidor** [comment]
@kiorky no problem, either way is fine. Thank you for the contribution! Will double check in a little bit and merge if all is ok.

File: Dockerfile [hunk 1]
```diff
@@ -6,11 +6,12 @@ EXPOSE 8090
 
 ADD build/libs/mxisd.jar /mxisd.jar
 ADD src/docker/start.sh /start.sh
-RUN mkdir -p /var/mxisd
+RUN apk update && apk add bash && rm -rf /var/lib/apk/* /var/cache/apk/* \
+    && mkdir -p /var/mxisd
```
Comment (maxidor) [inline Dockerfile]: Why `mkdir` needed? I made a container without this and it is created, regardless of me binding a host volume to it wiht `-v`.

File: src/docker/start.sh [hunk 2]
```diff
@@ -1,20 +1,22 @@
-#!/bin/sh
-
-if ! [ -z "$CONF_FILE_PATH" ] && ! [ -f "CONF_FILE_PATH" ]; then
+#!/usr/bin/env bash
+if [[ -n "$CONF_FILE_PATH" ]] && [ ! -f "CONF_FILE_PATH" ]; then
```
Comment (maxidor) [inline src/docker/start.sh]: Doesn't work. If the file already exists, it stills goes to `Generating config file`

File: src/docker/start.sh [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
     echo "Generating config file $CONF_FILE_PATH"
     touch "CONF_FILE_PATH"
 
-    if ! [ -z "$MATRIX_DOMAIN" ]; then
+    if [[ -n "$MATRIX_DOMAIN" ]] \
+      && ! egrep -q "^matrix.domain:" $CONF_FILE_PATH; then
```
Comment (maxidor) [inline src/docker/start.sh]: This should not be present if the above `if` works properly: nothing should be done on the config file if it already exists.

File: src/docker/start.sh [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
         echo "Setting matrix domain to $MATRIX_DOMAIN"
         echo "matrix.domain: $MATRIX_DOMAIN" >> "$CONF_FILE_PATH"
     fi
 
-    if ! [ -z "$SIGN_KEY_PATH" ]; then
+    if [[ -n "$SIGN_KEY_PATH" ]] \
+      && ! egrep -q "^key.path:" $CONF_FILE_PATH; then
```
Comment (maxidor) [inline src/docker/start.sh]: This should not be present if the above `if` works properly: nothing should be done on the config file if it already exists.

File: src/docker/start.sh [hunk 5]
```diff
@@ -1,20 +1,22 @@
...
         echo "Setting signing key path to $SIGN_KEY_PATH"
         echo "key.path: $SIGN_KEY_PATH" >> "$CONF_FILE_PATH"
     fi
 
-    if ! [ -z "$SQLITE_DATABASE_PATH" ]; then
+    if [[ -n "$SQLITE_DATABASE_PATH" ]] \
+      && ! egrep -q "^storage.provider.sqlite.database:" $CONF_FILE_PATH; then
```
Comment (maxidor) [inline src/docker/start.sh]: This should not be present if the above `if` works properly: nothing should be done on the config file if it already exists.

File: src/docker/start.sh [hunk 6]
```diff
@@ -1,20 +1,22 @@
-#!/bin/sh
-
-if ! [ -z "$CONF_FILE_PATH" ] && ! [ -f "CONF_FILE_PATH" ]; then
+#!/usr/bin/env bash
```
Comment (maxidor) [inline src/docker/start.sh]: why not just `#!/bin/bash` ?

Comment (kiorky) [reply src/docker/start.sh]: portability ...

**Comment by maxidor** [comment]
Cosmetic change to have this to my taste. Thank you for the contribution and fixing this!

**Pull request merged**